internal/tls/tls_bench_test.go

```diff
@@ -0,0 +1,66 @@
+package tls
+
+import (
+    gotls "crypto/tls"
+    "runtime"
+    "testing"
+
+    testdata "github.com/vdaas/vald/internal/test"
+)
+
+var certSink *gotls.Certificate
+
+func Benchmark_GetCertificate_Static(b *testing.B) {
+    cfg, err := NewServerConfig(
+        WithCert(testdata.GetTestdataPath("tls/server.crt")),
+        WithKey(testdata.GetTestdataPath("tls/server.key")),
+        WithServerCertHotReload(false),
+    )
+    if err != nil {
+        b.Fatalf("failed to create static server config: %v", err)
+    }
+    if len(cfg.Certificates) == 0 {
+        b.Fatalf("no static certificate loaded")
+    }
+    get := func() *gotls.Certificate { return &cfg.Certificates[0] }
+
+    _ = get()
+
+    b.ReportAllocs()
+    b.ResetTimer()
+    for i := 0; i < b.N; i++ {
+        certSink = get()
+    }
+    runtime.KeepAlive(certSink)
+}
+
+func Benchmark_GetCertificate_HotReload(b *testing.B) {
+    cfg, err := NewServerConfig(
+        WithCert(testdata.GetTestdataPath("tls/server.crt")),
+        WithKey(testdata.GetTestdataPath("tls/server.key")),
+        WithServerCertHotReload(true),
+    )
+    if err != nil {
+        b.Fatalf("failed to create hot-reload server config: %v", err)
+    }
+    if cfg.GetCertificate == nil {
+        b.Fatalf("GetCertificate not set for hot reload config")
+    }
+
+    chi := &gotls.ClientHelloInfo{ServerName: "bench"}
+
+    if c, err := cfg.GetCertificate(chi); err == nil {
+        certSink = c
+    }
+
+    b.ReportAllocs()
+    b.ResetTimer()
+    for i := 0; i < b.N; i++ {
+        c, err := cfg.GetCertificate(chi)
+        if err != nil {
+            b.Fatalf("GetCertificate failed: %v", err)
+        }
+        certSink = c
+    }
+    runtime.KeepAlive(certSink)
+}
```

Comment:
This benchmark is used differently from the actual case flow, and since it compares simple function calls with memory calls, it's obvious without benchmarking whether the HotReload process is slow.

The focus at this time should be building a simple Echo server in a Client-Server environment and accurately evaluating its QPS and latency using testing.B based code.